src/contracts/core/StakeRootCompendium.sol

```diff
@@ -26,170 +25,139 @@ contract StakeRootCompendium is StakeRootCompendiumStorage {
-            );
-            operatorSetToIndex[msg.sender][operatorSetId].push(uint32(block.timestamp), uint208(operatorSets.length));
-            operatorSets.push(operatorSet);
+        if (depositInfo[msg.sender][operatorSetId].balance < MIN_DEPOSIT_BALANCE) {
+            _removeOperatorSetFromStakeTree(operatorSet);
         }
 
-        // set the strategies and multipliers for the operator set
-        for (uint256 i = 0; i < strategiesAndMultipliers.length; i++) {
-            operatorSetToStrategyAndMultipliers[msg.sender][operatorSetId].set(
-                address(strategiesAndMultipliers[i].strategy), uint256(strategiesAndMultipliers[i].multiplier)
-            );
-        }
-        require(
-            operatorSetToStrategyAndMultipliers[msg.sender][operatorSetId].length() <= MAX_NUM_STRATEGIES, 
-            "StakeRootCompendium.setStrategiesAndMultipliers: too many strategies"
-        );
+        (bool success, ) = payable(msg.sender).call{value: amount}("");
+        require(success, "StakeRootCompendium.withdrawForOperatorSet: eth transfer failed");
 
-        _updateTotals(numStrategiesBefore, operatorSetToStrategyAndMultipliers[msg.sender][operatorSetId].length());
+        return amount;
     }
 
     /// @inheritdoc IStakeRootCompendium
     function removeStrategiesAndMultipliers(uint32 operatorSetId, IStrategy[] calldata strategies) external {
-        IAVSDirectory.OperatorSet memory operatorSet =
-            IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
         // update the deposit balance for the operator set whenever number of strategies is changed
-        _updateDepositBalanceInfo(operatorSet, true);
+        IAVSDirectory.OperatorSet memory operatorSet = IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId});
+        _updateDepositInfo(operatorSet);
+        uint256 numStrategiesBefore = operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].length();
+
+        for (uint256 i = 0; i < strategies.length; i++) {
+            require(
+                operatorSetToStrategyAndMultipliers[msg.sender][operatorSet.operatorSetId].remove(address(strategies[i])),
+                "StakeRootCompendium.removeStrategiesAndMultipliers: strategy not found"
+            );
+        }
+
+        _updateTotalStrategies(numStrategiesBefore, operatorSet);
+    }
+
 
-        // remove the strategies and multipliers for the operator set
-        _removeStrategiesAndMultipliers(operatorSet, strategies);
+    function removeOperatorSetFromStakeTree(uint32 operatorSetId) external {
+        _removeOperatorSetFromStakeTree(IAVSDirectory.OperatorSet({avs: msg.sender, operatorSetId: operatorSetId}));
     }
 
     /// @inheritdoc IStakeRootCompendium
     function setOperatorSetExtraData(uint32 operatorSetId, bytes32 extraData) external {
-        (bool exists,, uint224 index) = operatorSetToIndex[msg.sender][operatorSetId].latestCheckpoint();
```

Comment:
seems correct to revert here imo

Reply:
yeah i'll put that back, it's a setter so used very rarely